schemas/2018-01-01/Microsoft.EventGrid.json

```diff
@@ -60,11 +60,212 @@
       ],
       "description": "Microsoft.EventGrid/topics"
+    },
+    "eventSubscriptions": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.EventGrid/eventSubscriptions"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2018-01-01"
+          ]
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/EventSubscriptionProperties"
+            }
+          ],
+          "description": "Properties of the event subscriptions"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "description": "Microsoft.EventGrid/eventSubscription"
     }
   },
   "definitions": {
     "TopicProperties": {
       "description": "Properties of the Topic"
+    },
+    "EventSubscriptionProperties": {
+      "description": "Properties of the Event Subscription",
+      "type": "object",
+      "properties": {
+        "topic": {
+          "description": "Name of the topic of the event subscription.",
+          "type": "string",
+          "readOnly": true
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the event subscription.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Deleting",
+            "Succeeded",
+            "Canceled",
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true
+        },
+        "destination": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/WebHookEventSubscriptionDestination"
+            },
+            {
+              "$ref": "#/definitions/EventHubEventSubscriptionDestination"
+            }
+          ],
+          "description": "Information about the destination where events have to be delivered for the event subscription."
+        },
+        "filter": {
+          "$ref": "#/definitions/EventSubscriptionFilter",
+          "description": "Information about the filter for the event subscription."
+        },
+        "labels": {
+          "description": "List of user defined labels.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "destination"
+      ]
+    },
+    "EventSubscriptionDestination": {
+      "description": "Information about the destination for an event subscription",
+      "required": [
+        "endpointType"
+      ],
+      "type": "object",
+      "properties": {
+        "endpointType": {
+          "description": "Type of the endpoint for the event subscription destination",
+          "enum": [
+            "WebHook",
+            "EventHub"
+          ],
+          "type": "string"
+        }
+      },
+      "discriminator": "endpointType"
```

Comment:
Should the "discriminator" line be removed?

Reply:
Actually, we don't need "EventSubscriptionDestination" definition since "EventSubscriptionProperties " is directly referring to one of WebHookEventSubscriptionDestination or EventHubEventSubscriptionDestination. Removed the definition.